lib/matplotlib/ticker.py

```diff
@@ -2450,7 +2461,7 @@ def nonsingular(self, vmin, vmax):
             vmin, vmax = 1, 10  # Initial range, no data plotted yet.
         elif vmax <= 0:
             _api.warn_external(
-                "Data has no positive values, and therefore cannot be "
+                "Data has non-positive values, and therefore cannot be "
```

Comment:
```suggestion
                "Data has no positive values, and therefore cannot be "
```

I think this one should stay as "no positive" as in the other blocks we clip vmin to a small positive value.